Aqovia.Utilities.SagaMachine.Tests/StateMachineTests.cs

```diff
@@ -92,33 +91,23 @@ public StateMachineTests()
         [Fact(DisplayName = "Should not allow InitialiseState without Id")]
         public void SagaMachineShouldHaveInitialiseStateId()
         {
-            //Arrange
-            Guid thisId = Guid.NewGuid();
-
             _sagaMachine
                 .WithMessage<HelloMessage>((proccess, msg) => proccess
-                    .InitialiseState((hello) => new TestState
+                    .InitialiseState(hello => new TestState
                     {
                         SomeStateVariable = hello.Message
                     })
                     .Execute()
                 );
 
-            //Act
-
-            //Assert
-            Action shouldThrow = () =>
+            Func<Task> act = async () =>
             {
-
-                _sagaMachine.Handle(new HelloMessage
+                await _sagaMachine.Handle(new HelloMessage
                 {
                     Message = "Initialised"
-                }).Wait();
+                }).ConfigureAwait(false);
             };
-            shouldThrow.ShouldThrow<SagaException>();
-
-
-
+            act.ShouldThrow<SagaException>();
```

Comment:
Ah! Well I didn't realise that when I was figuring through it late night :-) Thanks

However, at least one of the tests uncovered a legitimate deadlock in the library itself, so I may need to add another test explicitly to deal with that.